### ISSUE TITLE
Instrument perf test cases in splinter_test to print extra diagnostics, elapsed-time monitoring. (Needed to resolve slow-execution time of this perf test.)

### ISSUE THREAD

**Issue by gapisback** [vmware/splinterdb#354]
To address slow running `splinter_test --perf` tests [ reported under #325 ], these test case methods have been refactored into sub-functions under issue #334. A new `--verbose` flag will be added under issue #353. 

In nightly CI-jobs, these tests never seem to complete for well over few hours. Even on Nimbus-VM runs, the `--perf` test is seen to take over 2 hours. (One example is logged under #314.) 

Here's a data point (dd. Sun 13.3.2022) from a most recent run off of /main @ SHA 8f192c1, run on machine `sdb-oss-test-vm`, with this h/w specs:

```
sdb-oss-test-vm:[27] $ osinfo.sh
Linux sdb-oss-test-vm 5.15.0-18-generic #18-Ubuntu SMP Fri Jan 21 14:57:54 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
 GenuineIntel, 32 CPUs, 62 GB,  Intel(R) Xeon(R) Gold 6138 CPU @ 2.00GHz
```
The nightly run tests run-time metrics are as follows:

```
Sun Mar 13 12:53:04 UTC 2022 **** SplinterDBNightlyTest Suite Execution Times ****

Functionality Stress test 10 mil rows, 1 tables, 4 GiB cache                    :   67s [  0h  1m  7s ]
Functionality Stress test 10 mil rows, 2 tables, 4 GiB cache                    :  137s [  0h  2m 17s ]
Functionality Stress test 1 mil rows, 2 tables, 1 MiB cache                     :   10s [  0h  0m 10s ]
Functionality Stress test 1 mil rows, 4 tables, 1 MiB cache                     :   24s [  0h  0m 24s ]
Performance (sync) test 8 insert, 8 lookup, 8 range lookup threads              : 7257s [  2h  0m 57s ]
Parallel Performance (sync) test 10 pthreads                                    : 1215s [  0h 20m 15s ]
Cache Performance test , default cache size                                     :   44s [  0h  0m 44s ]
Cache Performance test , 6 GiB cache                                            :  365s [  0h  6m  5s ]
Nightly Stress & Performance Tests                                              : 9151s [  2h 32m 31s ]
```

**The single long pole test** is : Performance (sync) test 8 insert, 8 lookup, 8 range lookup threads, taking ~ 3hrs.

```
506 + bin/driver_test splinter_test --perf --max-async-inflight 0 --num-insert-threads 8 --num-lookup-threads 8 --num-range-lookup-threads 8 --    lookup-positive-percent 10 --db-capacity-gib 60 --db-location splinter_test.perf.db
507 bin/driver_test: splinterdb_build_version 8f192c1c
508 Dispatch test splinter_test
[...]
549 per-splinter per-thread range time per tuple 118355832 ns
550 splinter total range rate: 67 ops/second
551 Statistics are not enabled
552 After destroy:
553 + set +x
554 test.sh: Performance (sync) test 8 insert, 8 lookup, 8 range lookup threads: 7257 s [ 2h 0m 57s ]
555
```
**We need extra timing metrics to see which phase(s) of the test is running slow.**

Under this issue, extend the messages and monitoring information collected for `--perf` and `--parallel-perf` test cases to display extra information, such as:

1. Improve the messaging of test execution parameters; e.g. # of rows, min / max ranges of queries etc.
2. Thread ID of threads executing
3. Progress of elapsed-times for key threads, and periodically for other threads, so we can see how progress is going on
4. Generate elapsed time metrics for each sub-phase of each test case

Basically, under this issue, instrument non-invasively extra timing / monitoring for these two test cases, to help troubleshoot the causes for the slowdown. (Actual resolution for the slow performance will come under follow-on issue, once we can pinpoint which phases are running slow, and by how much.)

**Comment by gapisback** [comment]
Some improved elapsed-time instrumentation under `--verbose-progress` flag has been implemented under:

```
5c99be6  (#354) splinter_test --perf: Fix bug in range-lookups test code.
```

Test has been included into set of nightly-run tests and is running cleanly now. Can close this issue out for now.
